File: CodeForces/1883b/main.cpp
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

void solve(){
	int n, m, c = 0;
	string s;
	cin >> n >> m >> s;

	int mp[26] = {0};

	for(int i = 0; i < s.size(); i++){
		mp[s[i]-'a']++;
	}

    for(int i = 0; i < 26; i++){
		if(mp[i] % 2 == 1){
			c++;
		}
	}
	if(c - m > 1){
		cout << "NO\n";
	}
	else {
		cout << "YES\n";
	}
    return;
}

int main() {
    int n;
    cin >> n;

    while(n--) {
        solve();
    }

    return 0;
}

File: CodeForces/1883b/main.py
def inp():
    return int(input())


def inlt():
    return list(map(int, input().split()))


def insr():
    return input().strip()


def solve():
    n, k = inlt()
    s = insr()
    odd = 0
    for i in set(s):
        odd += s.count(i) % 2
    if k + 1 < odd:
        print("NO")
    else:
        print("YES")


t = inp()
for _ in range(t):
    solve()
